doc/type_requirements/File.qbk

```diff
@@ -0,0 +1,183 @@
+[/
+    Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+[section:File File]
+
+The [*File] concept abstracts access to files in the underlying file system.
+
+To support interfaces other than Win32 or POSIX, callers may provide their
+own [*File] type that meets these requirements.
+
+[heading Associated Types]
+* [link nudb.ref.nudb__path_type path_type]
+* [link nudb.ref.nudb__file_mode file_mode]
+* [link nudb.ref.nudb__error_code error_code]
+
+[heading Notation]
+
+* `F` a [*File] type
+* `f` an instance of type F
+* `p` an instance of type [link nudb.ref.nudb__path_type path_type]
+* `m` an instance of the type [link nudb.ref.nudb__file_mode file_mode]
+* `ec` an instance of a reference to the type
+      [link nudb.ref.nudb__error_code error_code]
+* `n` a number of bytes. Type is `std::size_t`
+* 'o' an offset from the beginning of a file. Type is `std::size_t`
+* 'a' a memory address of type void *
+
+[heading Valid Expressions]
+[table Valid Expressions
+[[Operation] [Return Type] [Semantics, Pre/Post-conditions]]
+[
+    [`F()`]
+    [ ]
+    [
+        Default constructable
+    ]
+]
+[
+    [`F(f)`]
+    [ ]
+    [
+        Move constructable.  Note NOT copy constructable.
+    ]
+]
+
+[
+    [`f.is_open()`]
```

Comment:
Just wondering why you use `f` here, when the previous version used `c` indicating a possibly const value. In other words, do we no longer care about informing the reader that their `is_open` for user defined types may be `const`?